### DISCUSSION TITLE
Favourite Component/Button

### DISCUSSION THREAD

**Discussion by cindyebi** [visual-framework/vf-core#1654] [Proposals]
Following up on the [previous discussion](https://github.com/visual-framework/vf-core/discussions/1643) about new components for the Training Academy's Personal Space project, I thought it's better to separate them into different threads.

There is a need to have a custom component for **adding courses to favorites**. Although we have tried to use the vf-checkbox as seen on the mockup below to accommodate this need, the visual representation didn't seem to fit well with the goal of the action. A checkbox component better signifies an action of selecting multiple options rather than adding items to favorites. As seen on other websites such as Future Learn and Udemy, adding courses to favorite are visually signified as a _Star_ or _Heart_ icon button. 

Current mockup using vf-checkbox:
![Overview](https://user-images.githubusercontent.com/75834903/132227400-a1b43820-b3b3-4f50-9d7b-664844ca52a0.png)

Udemy:
![Screenshot 2021-09-06 at 15 45 13 (2)](https://user-images.githubusercontent.com/75834903/132234478-f9c3ab22-9e3b-410b-8114-1b14c28db750.png)

FutureLearn:
![Screenshot 2021-09-06 at 15 46 36](https://user-images.githubusercontent.com/75834903/132234499-702de842-cddd-49cb-a66e-d0c129c5884f.png)

Component design suggestions:
Similar to the _vf-back-to-top_, v1 and v2 added a _Star_ icon into the current vf-button secondary variant:
![Screenshot 2021-09-06 at 15 52 53](https://user-images.githubusercontent.com/75834903/132234975-d55a2692-fca9-4c44-aad2-25e61366a443.png)

An icon button without labels could be utilized for a more compact version where there isn't enough space to accommodate a button with label, e.g. course listings on search results, image thumbnails on the Featured Courses section on the Training Academy homepage:
![Screenshot 2021-09-06 at 15 52 59](https://user-images.githubusercontent.com/75834903/132235013-4b583823-18bf-444f-996e-24eb7420a23c.png)

**Comment by khawkins98** [comment]
Being worked on at https://github.com/visual-framework/vf-core/pull/1672
